try tbb-devel on windows

danielnachun opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

danielnachun committed:
try tbb-devel on windows

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

danielnachun committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.07.10.53.53

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

danielnachun committed:
patch makevars

danielnachun committed:
patch makevars

danielnachun committed:
patch makevars

danielnachun committed:
use r_clang

danielnachun committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2025.12.22.22.48.19

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

h-vetinari commented:
```
In file included from ../inst/include\RcppParallel.h:6:
../inst/include\RcppParallel/TinyThread.h:85:24: error: expected unqualified-id
   85 |       grainSize = std::max(length / threads, grainSize);
      |                        ^
C:\Program Files (x86)\Windows Kits\10\\include\10.0.26100.0\\shared\minwindef.h:193:29: note: expanded from macro 'max'
  193 | #define max(a,b)            (((a) > (b)) ? (a) : (b))
      |                             ^
```
You should add `-DNOMINMAX` to `CXXFLAGS` :)

Also note that you're picking up the compiler from the image
```
using C++ compiler: 'clang version 19.1.5'
```
c.f. https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/408

danielnachun committed:
set CXXFLAG and correct PATH

danielnachun committed:
delete MSVC Clang

danielnachun committed:
fix typo

danielnachun committed:
dump out Makevars

danielnachun committed:
fix typo

danielnachun committed:
fix typo

danielnachun committed:
fix typo

danielnachun committed:
use simplified Makevars.ucrt

danielnachun committed:
use simplified Makevars.ucrt

danielnachun committed:
tweak path

danielnachun commented:
@h-vetinari Thanks for the tip about the `CXXFLAG`, that fixed the compilation error.

The incorrect compiler issue ended up being a lot tricker than I first thought.  Once I deleted the compiler from the image, it was instead using the conda compiler but was bypassing the clang wrapper and directly calling `clang` which was failing because the clang wrapper has to remove `tmp.def` from the arguments.  

The root cause of the problem is actually that the `configure.win` script generates a much more complex `Makevars` file from `Makevars.in` that was somehow bypassing the clang wrapper.  The easiest fix for this ended up being to just create a simple `Makevars.ucrt` file and copying that into `src`.  All the extra complexity in the generated `Makevars` file was just for building the bundled TBB which we are not doing here, so the `Makevars.ucrt` file is actually quite simple.

Now the compilation and linking is working but there is some issue with an invalid escape character:
```
installing via 'install.libs.R' to D:/bld/r-rcppparallel_1766518091648/_h_env/lib/R/library/00LOCK-work/00new/RcppParallel
Error: '\_' is an unrecognized escape in character string (<input>:6:51)
* removing 'D:/bld/r-rcppparallel_1766518091648/_h_env/lib/R/library/RcppParallel'
```
